python_modules/libraries/dagster-aws/dagster_aws/ecs/executor.py

```diff
@@ -0,0 +1,394 @@
+    run_launcher = init_context.instance.run_launcher
+
+    check.invariant(
+        isinstance(run_launcher, EcsRunLauncher),
+        "Using the ecs_executor currently requires that the run be launched in an ECS task via the EcsRunLauncher.",
+    )
+
+    exc_cfg = init_context.executor_config
+
+    return StepDelegatingExecutor(
+        EcsStepHandler(
+            run_launcher=run_launcher,  # type: ignore
+            run_task_kwargs=exc_cfg.get("run_task_kwargs"),  # type: ignore
+            cpu=exc_cfg.get("cpu"),  # type: ignore
+            memory=exc_cfg.get("memory"),  # type: ignore
+            ephemeral_storage=exc_cfg.get("ephemeral_storage"),  # type: ignore
+            task_overrides=exc_cfg.get("task_overrides"),  # type:ignore
+        ),
+        retries=RetryMode.from_config(exc_cfg["retries"]),  # type: ignore
+        max_concurrent=check.opt_int_elem(exc_cfg, "max_concurrent"),
+        tag_concurrency_limits=check.opt_list_elem(exc_cfg, "tag_concurrency_limits"),
+        should_verify_step=True,
+    )
+
+
+@experimental
+class EcsStepHandler(StepHandler):
+    @property
+    def name(self):
+        return "EcsStepHandler"
+
+    def __init__(
+        self,
+        run_launcher: EcsRunLauncher,
+        run_task_kwargs: Optional[Mapping[str, Any]],
+        cpu: Optional[int],
+        memory: Optional[int],
+        ephemeral_storage: Optional[int],
+        task_overrides: Optional[Mapping[str, Any]],
+    ):
+        super().__init__()
+
+        run_task_kwargs = run_task_kwargs or {}
+
+        self.ecs = boto3.client("ecs")
+        self.ec2 = boto3.resource("ec2")
+
+        # confusingly, run_task expects cpu and memory value as strings
+        self._cpu = str(cpu) if cpu else None
+        self._memory = str(memory) if memory else None
+
+        self._ephemeral_storage = ephemeral_storage
+        self._task_overrides = check.opt_mapping_param(task_overrides, "task_overrides")
+
+        current_task_metadata = get_current_ecs_task_metadata()
+        current_task = get_current_ecs_task(
+            self.ecs, current_task_metadata.task_arn, current_task_metadata.cluster
+        )
+
+        if run_launcher.use_current_ecs_task_config:
+            current_task_kwargs = get_task_kwargs_from_current_task(
+                self.ec2,
+                current_task_metadata.cluster,
+                current_task,
+            )
+        else:
+            current_task_kwargs = {}
+
+        run_launcher_kwargs = {**current_task_kwargs, **run_launcher.run_task_kwargs}
+
+        self._cluster_arn = current_task["clusterArn"]
+        self._task_definition_arn = current_task["taskDefinitionArn"]
+
+        # note: more kwargs will be pulled from the run launcher
+        # during launch_step
+        self._run_task_kwargs = {
+            **run_launcher_kwargs,
+            **run_task_kwargs,
+            "taskDefinition": current_task["taskDefinitionArn"],
```

Comment:
if the user has container_context.task_definition_arn set I think we want to use that here instead

I also think this order might not be quite right? This should be first in the list so that you can override it via config if you want

Reply:
> if the user has container_context.task_definition_arn set I think we want to use that here instead

This will come from `run_launcher_kwargs` right?

Reply:
We don't know the container context yet, it is run-dependant - so it's not actually possible for it to have it here yet. We would need to check it and set it later.

the relevant code from the run launcher is here: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py#L661-L664